tools/importer-rest-api-specs/components/parser/dataworkarounds/workaround_datafactory_28122.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package dataworkarounds
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/pandora/tools/data-api-sdk/v1/models"
+	importerModels "github.com/hashicorp/pandora/tools/importer-rest-api-specs/models"
+)
+
+var _ workaround = workaroundDataFactory28122{}
+
+// TODO swagger PR link - this be added when the issue has been opened
+type workaroundDataFactory28122 struct{}
+
+func (workaroundDataFactory28122) IsApplicable(apiDefinition *importerModels.AzureApiDefinition) bool {
+	return apiDefinition.ServiceName == "DataFactory" && apiDefinition.ApiVersion == "2018-06-01"
+}
+
+func (workaroundDataFactory28122) Name() string {
+	return "DataFactory / 28122"
+}
+
+func (workaroundDataFactory28122) Process(apiDefinition importerModels.AzureApiDefinition) (*importerModels.AzureApiDefinition, error) {
+	resource, ok := apiDefinition.Resources["Credentials"]
+	if !ok {
+		return nil, fmt.Errorf("couldn't find API Resource Credentials")
+	}
+
+	// add the new discriminated parent type
+	managedIdentityCredential, ok := resource.Models["ManagedIdentityCredential"]
+	if !ok {
+		return nil, fmt.Errorf("couldn't find model ManagedIdentityCredential")
+	}
+
+	managedIdentityCredential.DiscriminatedValue = pointer.To("ManagedIdentity")
+	managedIdentityCredential.FieldNameContainingDiscriminatedValue = pointer.To("Type")
+	managedIdentityCredential.ParentTypeName = pointer.To("Credential")
+
+	managedIdentityCredential.Fields["TypeProperties"] = models.SDKField{
+		ContainsDiscriminatedValue: false,
+		JsonName:                   "typeProperties",
+		ObjectDefinition: models.SDKObjectDefinition{
+			Type:          models.ReferenceSDKObjectDefinitionType,
+			ReferenceName: pointer.To("ManagedIdentityTypeProperties"),
```

Comment:
_minor_ since we're referencing this model, it might be worth validating that this Model exists here? If not it'd get caught later in the process if not, but it'd be clearer here as to what's failed without digging in so deeply?